### PR TITLE
[CI] Run unit tests in pipeline

### PR DESCRIPTION
**Pull request by nqb**
# Description
- Replace "install" stage by "test" stage in pipeline to run unit tests (Perl and Golang) inside Vagrant virtual machines
- Refactorize Ansible inventory to less configure `pf*dev` machines: more tests will be done during integration tests
- Documentation to run unit tests on local machines
- Changes in `t/` directory to run unit tests using MariaDB instance listening on localhost (default installation made by PF)

# Impacts
Duration of pipeline if jobs are run on each commit.

# Delete branch after merge
YES

# Checklist
- [X] Document the feature

# NEWS file entries

## Enhancements
* CI: run unit tests in pipeline

### CODE REVIEW

**Review by julsemaan — APPROVED** [review]

**Comment by nqb** [comment]
Please wait before merging because Ansible collection has not been uploaded to galaxy.ansible.com due to https://github.com/ansible/galaxy/issues/2173

**Comment by julsemaan** [comment]
When a PR is not ready to be merged, mark it as not ready. That way we won't merge it or review it until you're ready

**Comment by extrafu** [comment]
@jrouzierinverse Review and merge ASAP.

File: ci/lib/test/run-tests.sh
Comment (jrouzierinverse) [inline ci/lib/test/run-tests.sh]: set the default to no

File: ci/lib/test/run-tests.sh
Comment (jrouzierinverse) [inline ci/lib/test/run-tests.sh]: set the default to no

**Comment by nqb** [comment]
@jrouzierinverse, I updated defaults value as requested.

**Comment by nqb** [comment]
@jrouzierinverse, do you want me to adjust other things ? If not, could you merge ? Thanks.

**Pull request merged**
